src/fsql/query.py

```diff
@@ -94,3 +96,66 @@ def eval_available(self, columns: dict[str, str]) -> bool:
 
 Q_TRUE = ConstantQuery(True)
 Q_FALSE = ConstantQuery(False)
+
+
+class ColumnComparator(Enum):
+    lex = auto()
+    num = auto()
+    wld = auto()
+
+    def compare(self, a: str, b: str) -> int:
+        if self is ColumnComparator.wld:
+            return 0
+        elif self is ColumnComparator.num:
+            return int(a) - int(b)
+        elif self is ColumnComparator.lex:
+            if a < b:
+                return -1
+            elif a == b:
+                return 0
+            else:
+                return 1
+
+
+@dataclass
+class ColumnRange:
+    name: str
+    min_value: str
+    max_value: str
+    column_comparator: ColumnComparator = ColumnComparator.lex
+
+
+class SimpleRangeQuery(Query):
+    """This is a query to return all files that lie >= c1=s1/c2=s2/... but < c1=e1/c2=e2/...
+    Usage of generators is prefered if the range can be explicitly generated (e.g., for Date Columns), but
+    in general case this does the job as well. ColumnComparator can be used to distinguish between:
+         - wld: a wildcard, any value in the given column satisfies but comparison continues,
+         - num: values are treated numerically, that is, 9 < 10. However, all partition values must be `int()`,
+         - lex: the default, values are compared lexicographically.
+
+    See tests/test_simple_range_query.py for examples"""
+
+    def __init__(self, ranges: list[ColumnRange]):
+        self.ranges = ranges
+
+    def _eval_generic(self, columns: dict[str, str], on_early_stop: bool) -> bool:
+        for c in self.ranges:
+            if c.name not in columns:
+                return on_early_stop
+            if c.column_comparator == ColumnComparator.wld:
+                continue
+            value = columns[c.name]
+            left = c.column_comparator.compare(c.min_value, value)
+            right = c.column_comparator.compare(value, c.max_value)
+            if left < 0 and right < 0:
+                return True
```

Comment:
`continue`?

Reply:
nn -- in the lexicographical comparison, you know that the following columns cannot get you out of the range anymore